homeassistant/components/shelly/coordinator.py

```diff
@@ -426,6 +427,19 @@ def _unsubscribe() -> None:
 
         return _unsubscribe
 
+    @callback
+    def async_subscribe_input_events(
+        self, input_event_callback: Callable[[dict[str, Any]], None]
+    ) -> CALLBACK_TYPE:
+        """Subscribe to input events."""
+
+        def _unsubscribe() -> None:
```

Comment:
How come this function isn't in line with the callable[dict[str, any]] declared above? Is this how python works?

Reply:
`_unsuscribe()` is the function that is returned by `async_subscribe_input_events()` so its type is `CALLBACK_TYPE` and `CALLBACK_TYPE` is `Callable[[], None]`.

Reply:
Oooh, I looked the wrong way. You're right! Thanks